File: app/src/main/java/com/team2052/frckrawler/activities/EventInfoActivity.java
package com.team2052.frckrawler.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;

import com.team2052.frckrawler.R;
import com.team2052.frckrawler.adapters.tab.EventViewPagerAdapter;
import com.team2052.frckrawler.views.LockableFloatingActionButton;

/**
 * @author Adam
 * @since 10/16/2014
 */
public class EventInfoActivity extends DatabaseActivity implements View.OnClickListener {
    ViewPager mViewPager;
    TabLayout mTabLayout;

    private EventViewPagerAdapter mAdapter;
    private LockableFloatingActionButton mFab;

    public static Intent newInstance(Context context, long event_id) {
        Intent intent = new Intent(context, EventInfoActivity.class);
        intent.putExtra(PARENT_ID, event_id);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        long mEvent_id = getIntent().getLongExtra(PARENT_ID, 0);
        setContentView(R.layout.layout_tab_fab);
        mViewPager = (ViewPager) findViewById(R.id.view_pager);
        mTabLayout = (TabLayout) findViewById(R.id.tab_layout);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);

        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        mFab = (LockableFloatingActionButton) findViewById(R.id.floating_action_button);
        mFab.setOnClickListener(this);

        mViewPager.setAdapter(mAdapter = new EventViewPagerAdapter(getSupportFragmentManager(), mEvent_id));
        mTabLayout.setupWithViewPager(mViewPager);

        mViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
            }

            @Override
            public void onPageSelected(int position) {
                if (position < 2) {
                    mFab.hide();
                    mFab.setLocked(true);
                } else {
                    mFab.setLocked(false);
                    mFab.show();
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {
            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void inject() {
        getComponent().inject(this);
    }

    @Override
    public void onClick(View v) {
        mAdapter.onClick(v, mViewPager.getCurrentItem());
    }
}

File: app/src/main/java/com/team2052/frckrawler/activities/EventInfoActivity.kt
package com.team2052.frckrawler.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.view.ViewPager
import android.view.MenuItem
import android.view.View
import com.team2052.frckrawler.R
import com.team2052.frckrawler.adapters.tab.EventViewPagerAdapter
import com.team2052.frckrawler.bindToViewPagerAndTabLayout
import kotlinx.android.synthetic.main.layout_tab_fab.*

/**
 * @author Adam
 * *
 * @since 10/16/2014
 */
class EventInfoActivity : DatabaseActivity(), View.OnClickListener {
    var adapter: EventViewPagerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val mEvent_id = intent.getLongExtra(DatabaseActivity.PARENT_ID, 0)
        setContentView(R.layout.layout_tab_fab)

        setSupportActionBar(toolbar)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        floating_action_button.setOnClickListener(this)

        adapter = EventViewPagerAdapter(supportFragmentManager, mEvent_id)
        adapter?.bindToViewPagerAndTabLayout(view_pager, tab_layout)

        view_pager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(p0: Int, p1: Float, p2: Int) {}
            override fun onPageScrollStateChanged(state: Int) {}

            override fun onPageSelected(position: Int) {
                if (position < 3) {
                    floating_action_button.hide()
                    floating_action_button.locked = true
                } else {
                    floating_action_button.locked = false
                    floating_action_button.show()
                }
            }
        })
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun inject() {
        component.inject(this)
    }

    override fun onClick(v: View) {
        adapter?.onClick(v, view_pager.currentItem)
    }

    companion object {

        fun newInstance(context: Context, event_id: Long): Intent {
            val intent = Intent(context, EventInfoActivity::class.java)
            intent.putExtra(DatabaseActivity.PARENT_ID, event_id)
            return intent
        }
    }
}
